src/modules/vision_target_estimator/Orientation/KF_orientation_moving.h

```diff
@@ -0,0 +1,114 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2013-2018 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file KF_orientation_moving.h
+ * @brief Filter to estimate the orientation of moving targets. State: [theta, theta_dot]
+ *
+ * @author Jonas Perolini <jonspero@me.com>
+ *
+ */
+
+#include <matrix/math.hpp>
+#include <mathlib/mathlib.h>
+#include <matrix/Matrix.hpp>
+#include <matrix/Vector.hpp>
+
+#include "base_KF_orientation.h"
+
+#pragma once
+
+namespace vision_target_estimator
+{
+class KF_orientation_moving : public Base_KF_orientation
```

Comment:
Shouldn't be `Base_KF_orientation` a member of `KF_orientation_moving` instead of a parent? I don't see why this class should inherit from the orientation KF.

Reply:
Improved for the position filter in https://github.com/PX4/PX4-Autopilot/pull/22008/commits/53b45be8e28c1f35870dbc7e2a0f8ab3f509d0ac. Still work to be done in the Orientation filter